File: examples/typescript/src/send.ts
// send.ts — send a single SMS

import { sendSMS } from './firesms.js';
import { isSuccess } from './types.js';

const API_KEY = process.env.FIRESMS_API_KEY;

if (!API_KEY) {
  console.error('Error: FIRESMS_API_KEY environment variable is not set.');
  process.exit(1);
}

try {
  const result = await sendSMS({
    apiKey: API_KEY,
    to: '27821234567',
    text: 'Hello from Fire SMS via TypeScript! 🔥',
  });

  if (isSuccess(result)) {
    console.log('✅ SMS sent successfully!');
    console.log('   Message ID:', result.id);
  } else {
    console.error('❌ Failed to send SMS:', result.error);
  }
} catch (err: unknown) {
  const message = err instanceof Error ? err.message : String(err);
  console.error('❌ Request failed:', message);
}

File: examples/python/send.py
# send.py — send a single SMS

from firesms import send_sms, FireSMSError

try:
    result = send_sms(
        to='27821234567',
        text='Hello from Fire SMS via Python! 🔥',
    )
    print('✅ SMS sent successfully!')
    print('   Message ID:', result['id'])

except FireSMSError as e:
    print('❌ API error:', e)

except Exception as e:
    print('❌ Request failed:', e)
